jax/_src/interpreters/mlir.py

```diff
@@ -1172,9 +1172,13 @@ def check_jaxpr_constants(closed_jaxpr: core.ClosedJaxpr):
   if (threshold := config.captured_constants_warn_bytes.value) == -1:
     return
 
+  if config.use_simplified_jaxpr_constants.value:
```

Comment:
Can't we just delete this function since closed over constants won't exist after your change?

Reply:
I agree. With the new handling of constants I cannot think of a strong reason why would closing over constants be bad, or worse than passing constants as args.

As a contrived example, if the user passes the constant as args it will force the user to think how it is sharded, and how it is handled by transformations. This is more work for the user but it may be something that the user ought to think about. But I don't know if this is a strong reason to keep the warning.

Zac, do you agree that we can retire this check when we enable the new handling of constants.

Reply:
I understand that the new captured constants handling stages these values out when they are values of type `jax.Array`, but I think there's plenty of room for users to create `np.ndarray` constants that, if I understand correctly, can still be captured. In fact a particular case of capturing 130GB of constants as NumPy arrays was the motivation for the check in the first place.

Reply:
Indeed, this covers ndarray also. And the HLO will does not contain the captured constants anymore.

Reply:
Instead of abandoning the change, you can only run it if the flag is off.

Then when removing the flag, it will be easy to remember which code to delete.